src/fund/components/FundCardAmountInput.tsx

```diff
@@ -0,0 +1,197 @@
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: We only want to focus the input when the input type changes
+  useEffect(() => {
+    // focus the input when the input type changes
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [inputType]);
+
+  return (
+    <div className="flex py-6">
+      <style>
+        {`
+          input[type="number"]::-webkit-inner-spin-button,
+          input[type="number"]::-webkit-outer-spin-button {
+            -webkit-appearance: none;
+            appearance: none;
+            margin: 0;
+          }
+
+          input[type="number"] {
+            -moz-appearance: textfield;
+          }
+        `}
+      </style>
+
+      <div className="flex" style={{ height: '78px' }}>
+        {/* Display the fiat currency sign before the input*/}
+        {inputType === 'fiat' && currencySign && (
+          <FundCardCurrencyLabel
+            ref={currencySpanRef}
+            currencySign={currencySign}
+          />
+        )}
+
+        <input
+          className={cn(
+            componentTheme,
+            text.body,
+            'border-[none] bg-transparent',
+            'text-[60px] leading-none outline-none',
+          )}
+          type="number"
+          value={value}
+          onChange={handleChange}
+          ref={inputRef}
+          inputMode="decimal"
+          minLength={1}
+          placeholder="0"
+          data-testid="ockFundCardAmountInput"
+        />
+        {/* Display the crypto asset symbol after the input*/}
+        {inputType === 'crypto' && assetSymbol && (
+          <FundCardCurrencyLabel
+            ref={currencySpanRef}
+            currencySign={assetSymbol}
+          />
+        )}
+      </div>
+
+      {/* Hidden span for measuring text width 
+          Without this span the input field would not adjust its width based on the text width and would look like this:
+          [0.12--------Empty Space-------][ETH] - As you can see the currency symbol is far away from the inputed value
+
+          With this span we can measure the width of the text in the input field and set the width of the input field to match the text width
+          [0.12][ETH] - Now the currency symbol is displayed next to the input field
+      */}
+      <span
+        data-testid="ockHiddenSpan"
+        ref={hiddenSpanRef}
+        className={cn(
+          componentTheme,
+          text.body,
+          'border-[none] bg-transparent',
+          'text-[60px] leading-none outline-none',
+        )}
+        style={{
+          position: 'absolute',
+          opacity: 0,
+          whiteSpace: 'nowrap',
+          pointerEvents: 'none',
+        }}
+      >
+        {value ? `${value}.` : '0.'}
+      </span>
+    </div>
+  );
+};
+
+export default FundCardAmountInput;
+
+/**
+ * Ensure the decimal value is formatted correctly (i.e. "0.1" instead of ".1" and "0.1" instead of "01")
+ */
+const formatDecimalInputValue = (value: string) => {
```

Comment:
This component is getting pretty complex - I feel like we should try to export some functionality into utility functions.